packages/jsii-pacmak/lib/targets/golang/module.ts

```diff
@@ -1,64 +1,254 @@
 import { CodeMaker } from 'codemaker';
-import { Assembly, Submodule as JsiiSubmodule } from 'jsii-reflect';
+import { Assembly, Type, Submodule as JsiiSubmodule } from 'jsii-reflect';
+import { join } from 'path';
 import { GoClass, Enum, Interface } from './types';
 
-export interface ModuleTypes {
-  [fqn: string]: Interface | Enum | GoClass;
+type ModuleType = Interface | Enum | GoClass;
+type ModuleTypes = ModuleType[];
+
+// TODO: Make this a class?
+export interface Module {
+  root: RootModule;
+  packageName: string;
+  moduleName: string;
+  filePath: string;
+  file: string;
+  submodules: Submodule[];
+  types: ModuleTypes;
+  hasSubmodules(): boolean;
+}
+
+function buildSubmodules(
+  root: RootModule,
+  parent: Module,
+  submodules: readonly JsiiSubmodule[],
+): Submodule[] {
+  return submodules.map((sm) => new Submodule(root, parent, sm));
 }
 
-export class Module {
-  public readonly types: ModuleTypes = {};
-  public readonly submodules: Submodule[];
-  public readonly assembly: Assembly | JsiiSubmodule;
+function unknownType(type: Type): never {
+  throw new Error(`Type: ${type.name} is not an interface, enum, or class`);
```

Comment:
Should we keep this error message more generic and simply say "`Type: ${type.name} is not a supported type`?